oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -34,8 +35,8 @@ pub use channel::{Handle, HandleDirection};
 
 struct Node {
     #[allow(dead_code)]
-    reference: NodeRef,
-    instance: Box<dyn crate::node::Node>,
+    reference: NodeId,
+    instance: Option<Box<dyn crate::node::Node>>,
```

Comment:
When stopping we can't empty the `runtime.nodes` and then call stop anymore because then all channels will become invalid as the runtime is checking accessibility via `runtime.nodes`. Instead i've made the instance an option so that during `runtime.stop` we can switch out only the `instance` leaving `runtime.nodes`, and then call stop on the `instances` that we have moved into a separate vec.

Reply:
This seems like it would be a good comment to have on the field itself :slightly_smiling_face: